roofit/roofitcore/src/RooSimultaneous.cxx

```diff
@@ -1060,7 +1060,7 @@ RooFit::OwningPtr<RooDataSet> RooSimultaneous::generateSimGlobal(const RooArgSet
       std::unique_ptr<RooDataSet> tmp{pdftmp->generate(globtmp,1)};
 
       // Transfer values to output placeholder
-      globClone.assign(*tmp->get(0)) ;
+      if (tmp->get(0)) globClone.assign(*tmp->get(0)) ;
```

Comment:
The dataset `tmp` is explicitly generated to have one event:
```c++
std::unique_ptr<RooDataSet> tmp{pdftmp->generate(globtmp,1)};
```
So the `if` you introduced should not be necessary. I'll have to take a closer look at the code in the upcoming week.

Thanks a ton for what you did so far, doing the initial push to resolve the issue!

Reply:
You are welcome.

> The dataset `tmp` is explicitly generated to have one event:

Apparently this is not always the case, in case some error is found downstream, resulting tmp is empty.